### ISSUE TITLE
react-powerplug breaking build after babel 7 upgrade

### ISSUE THREAD

**Issue by dearcodes** [renatorib/react-powerplug#166]
Hi!

Using version `1.0.0-rc.1` breaks the build when upgrading to babel 7.

`Module not found: Error: Can't resolve '@babel/runtime/helpers/builtin/es6/objectWithoutProperties' in '.../node_modules/react-powerplug/dist'`

Any clue what is happening?

**Comment by TrySound** [comment]
Do you have resolution with `@babel/runtime` or do you use next.js/create-react-app?
This dependency is fixed and should be loaded correctly until you break normal flow
https://unpkg.com/react-powerplug@1.0.0-rc.1/package.json

**Comment by danielreyesGL** [comment]
Yes, I'm using `"@babel/runtime": "^7.0.0"`

Maybe the problem is related to you guys having `"@babel/runtime": "7.0.0-beta.49"` package resolution?

Version mismatch?

**Comment by dearcodes** [comment]
@TrySound Yes, I'm using `"@babel/runtime": "^7.0.0"`

Maybe the problem is related to you guys having `"@babel/runtime": "7.0.0-beta.49"` package resolution?

Version mismatch?

**Comment by dearcodes** [comment]
My current installed version of `@babel/runtime` is 7.0.0. 
`react-powerplug` published library relies on `7.0.0-beta.49` which has conflicts with current latest release of `@babel/runtime`

**Comment by TrySound** [comment]
@dearcodes As I said the dependency version is pinned. This means package manager should keep old version as a sub module in powerplug directory.

The problem may also happen if you use yarn flat mode. Please, don't. Package manager hoists modules if module can be hoisted.

**Comment by lloydh** [comment]
I was also seeing this on a Gatsby v2 development site. Same error with yarn and npm.

Downgrading to the 0.1.x release got things working but a fix would be good.

**Comment by TrySound** [comment]
Guys, could you check do you have your_project_dir/node_modules/react-powerplug/node_modules/@babel/runtime directory?

**Comment by lloydh** [comment]
Switched back to 1.0.0-rc.1, did a fresh dependency install (yarn) and the `react-powerplug/node_modules/@babel/runtime` directory is present (`7.0.0-beta.49`).

The error persists though:
```
These dependencies were not found:

* @babel/runtime/helpers/builtin/es6/assertThisInitialized in ./node_modules/react-powerplug/dist/react-powerplug.esm.js
* @babel/runtime/helpers/builtin/es6/defineProperty in ./node_modules/react-powerplug/dist/react-powerplug.esm.js
* @babel/runtime/helpers/builtin/es6/inheritsLoose in ./node_modules/react-powerplug/dist/react-powerplug.esm.js
* @babel/runtime/helpers/builtin/es6/objectSpread in ./node_modules/react-powerplug/dist/react-powerplug.esm.js
* @babel/runtime/helpers/builtin/es6/objectWithoutProperties in ./node_modules/react-powerplug/dist/react-powerplug.esm.js
```

**Comment by lloydh** [comment]
Looking at the output in the esm.js entrypoint (`react-powerplug/dist/react-powerplug.esm.js`), won't these imports reference the project dependencies (as opposed to `../node_modules/@babel/runtime/*`)?

```javascript
import _objectSpread from '@babel/runtime/helpers/builtin/es6/objectSpread';
import _inheritsLoose from '@babel/runtime/helpers/builtin/es6/inheritsLoose';
import _assertThisInitialized from '@babel/runtime/helpers/builtin/es6/assertThisInitialized';
import _defineProperty from '@babel/runtime/helpers/builtin/es6/defineProperty';
import _objectWithoutProperties from '@babel/runtime/helpers/builtin/es6/objectWithoutProperties';
```

If so, that's consistent with the error messages. Sorry I'm not familiar enough with rollup/babel to review the config.

**Comment by TrySound** [comment]
It's not about rollup/babel. Node resolver which is used for rollup/webpack/parcel and other resolves nested dependency first and then looks for toplevel dependency.

**Comment by damassi** [comment]
Also running into this issue with storybooks v4

**Comment by damassi** [comment]
The structure of the `@babel/runtime` folder has changed and has flattened out: 

<img width="754" alt="screen shot 2018-10-11 at 8 49 53 pm" src="https://user-images.githubusercontent.com/236943/46847061-48cdc200-cd97-11e8-8837-5fbdafdbf2ed.png">

**Comment by DeX3** [comment]
Any updates on this? Is there a way to make powerplug work with yarn?

**Comment by cheapsteak** [comment]
It's a problem with Storybook v4's webpack config aliasing `@babel/runtime` to the version that's in the root project's dependencies

https://github.com/storybooks/storybook/blob/a9e8e42744721144063d2c9e03ec25bc49682d08/lib/core/src/server/config/webpack.config.dev.js#L101-L103

**Comment by renatorib** [comment]
I'll close this issue since it's a Storybook issue. If we get into trouble again, I'll reopen here.

**Comment by kserjey** [comment]
There was a problem with beta version of `@babel/runtime`, but it's gone when you've released 1.0.0. Hooray!
